<h1><center> Analysis of urban versus rural living</center></h1>
![](https://sustainabilityatspu.files.wordpress.com/2014/01/urban-and-rural.jpg?w=1097&h=784)

In [1]:
%matplotlib inline
import pandas as pd
import copy
import matplotlib.pyplot as plt
import matplotlib.mlab as mlab
import numpy as np
import seaborn as sns
from scipy import stats

#import corrplot

# styling
pd.set_option('display.max_columns',150)
plt.style.use('bmh')
from IPython.display import display

import warnings
warnings.filterwarnings("ignore")

In [2]:
young = pd.read_csv('../../datasets/from_our_childhood.csv')

En el siguiente informe, presentaremos un análisis de la independencia entre las variables 'Smoking' y 'Alcohol'. Para la misma se realizará un test de chi cuadrado con las siguientes hipótesis:
   * $H_0$ = 'Fumar y tomar alcohol son dos variables independientes'
   * $H_1$ = 'Existe dependencia entre fumar y tomar alcohol'
   
Se intentará probar que existe una dependencia, es decir, que $H_1$ es válida. Para esto primero se procederá a analizar la tabla de frecuencia observadas y marginales para dichas variables.

In [5]:
pd.crosstab(young.Alcohol, young.Smoking, margins = True)

Smoking         current smoker  former smoker  never smoked  tried smoking  \
Alcohol                                                                      
drink a lot                 78             55            15             72   
never                       11             11            61             39   
social drinker              99            109           131            319   
All                        188            175           207            430   

Smoking          All  
Alcohol               
drink a lot      220  
never            122  
social drinker   658  
All             1000

Es posible observar en la tabla que la gran mayoría de los fumadores activos bebe. Del total de estos, un **41%** bebe mucho y un **52%** es un bebedor social reflejando que el **93%** de la gente que fuma activamente bebe.

Lo mismo se puede observar entre las personas que intentaron fumar. De esta categoría el **91%** bebe mucho o es un bebedor social lo cual nos lleva a inferir que hay una relación entre ambas variables.

Para poder realizar una prueba de independencia de chi cuadrado es necesario además tener una tabla de las frecuencias esperadas. Mientras más se acerquen los valores observados a los esperados el valor del estadístico se acercara a 0 avalando $H_0$. Caso contrario, si los valores son muy dispares se rechazaría la hipótesis nula.

In [46]:
# Observed frecuency
observed_frecuency = pd.crosstab(young.Alcohol, young.Smoking)

# Calculate the chi-square results through scipy special function 'chi2_contigency'
chi_square_results = stats.chi2_contingency(observed_frecuency)
[value, p_value, g, frequency_array] = chi_square_results
smoking_labels = ['current smoker', 'former smoker', 'never smoked', 'tried smoking']
alcohol_labels = ['drink a lot', 'never', 'social drinker']


# Expected frecuency
expected_frecuency = pd.DataFrame(frequency_array, columns=smoking_labels, index=alcohol_labels)
# Add total per row and column
expected_frecuency['All'] = expected_frecuency.sum(axis=1)
expected_frecuency.loc['All']= expected_frecuency.sum()
expected_frecuency

current smoker  former smoker  never smoked  tried smoking  \
drink a lot             41.360          38.50        45.540          94.60   
never                   22.936          21.35        25.254          52.46   
social drinker         123.704         115.15       136.206         282.94   
All                    188.000         175.00       207.000         430.00   

                   All  
drink a lot      220.0  
never            122.0  
social drinker   658.0  
All             1000.0

In [51]:
print('estadístico chi cuadrado (experimental): {}\n'.format(value))
print('p-valor: {}\n'.format(p_value))
print('Grados de libertad: {}\n'.format(g))

estadístico chi cuadrado (experimental): 140.7461035508088

p-valor: 6.974221284156533e-28

Grados de libertad: 6



Si comparamos el valor del estadístico chi-cuadrado experiemental con el valor crítico del modelo veremos que si tomamos un $ \alpha = 0.05 $ con **6** grados de libertad

 $ \chi^2_{experimental}\  \gt \  \chi^2_{6, 0.05} $
 
 $ 140.746\  \gt \ 12.592 $
 
 
 Lo cual nos lleva la conclusión de que debemos rechazar la hipótesis nula por existir dependencia entre ambas variables. 
#### Haría falta un análisis del p-valor y otras opciones para alpha